cpp/tests/spatial/polygon_distance_test.cpp

```diff
@@ -0,0 +1,152 @@
+{
+  using T = TypeParam;
+
+  auto x             = fixed_width_column_wrapper<T>({2});
+  auto y             = fixed_width_column_wrapper<T>({2});
+  auto space_offsets = fixed_width_column_wrapper<size_type>({0});
+
+  auto expected = fixed_width_column_wrapper<T>({0.0});
+  auto actual   = cuspatial::directed_polygon_distance(x, y, space_offsets);
+
+  expect_columns_equivalent(expected, actual->view(), true);
+}
+
+TYPED_TEST(DirectedPolygonDistanceTest, TwoShapesPointToPoint)
+{
+  using T = TypeParam;
+
+  auto x             = fixed_width_column_wrapper<T>({-1, -2, -3, 1, 2, 3});
+  auto y             = fixed_width_column_wrapper<T>({-1, -3, -2, 1, 3, 2});
+  auto space_offsets = fixed_width_column_wrapper<size_type>({0, 3});
+
+  auto expected = fixed_width_column_wrapper<T>({0.0, 2.8284271247461903, 2.8284271247461903, 0.0});
+  auto actual   = cuspatial::directed_polygon_distance(x, y, space_offsets);
+
+  expect_columns_equivalent(expected, actual->view(), true);
+}
+
+TYPED_TEST(DirectedPolygonDistanceTest, EdgesOnly)
+{
+  using T = TypeParam;
+
+  auto x             = fixed_width_column_wrapper<T>({0, 0, 0, 0});
+  auto y             = fixed_width_column_wrapper<T>({0, 0, 0, 0});
+  auto space_offsets = fixed_width_column_wrapper<size_type>({0, 2});
+
+  auto expected = fixed_width_column_wrapper<T>({0.0, 0.0, 0.0, 0.0});
+  auto actual   = cuspatial::directed_polygon_distance(x, y, space_offsets);
+
+  expect_columns_equivalent(expected, actual->view(), true);
+}
+
+TYPED_TEST(DirectedPolygonDistanceTest, ZeroLengthEdge)
+{
+  using T = TypeParam;
+
+  auto x             = fixed_width_column_wrapper<T>({0, 0});
+  auto y             = fixed_width_column_wrapper<T>({0, 0});
+  auto space_offsets = fixed_width_column_wrapper<size_type>({0});
+
+  auto expected = fixed_width_column_wrapper<T>({0});
+  auto actual   = cuspatial::directed_polygon_distance(x, y, space_offsets);
+
+  expect_columns_equal(expected, actual->view(), true);
+}
+
+TYPED_TEST(DirectedPolygonDistanceTest, MismatchedTypeTest)
+{
+  auto x             = fixed_width_column_wrapper<double>({0, 0});
+  auto y             = fixed_width_column_wrapper<float>({0, 0});
+  auto space_offsets = fixed_width_column_wrapper<size_type>({0});
+
+  EXPECT_THROW(cuspatial::directed_polygon_distance(x, y, space_offsets), cuspatial::logic_error);
+}
+
+template <typename T>
+struct DirectedPolygonDistanceUnsupportedTypesTest : public BaseFixture {
+};
+
+TYPED_TEST_CASE(DirectedPolygonDistanceUnsupportedTypesTest, UnsupportedTestTypes);
+
+TYPED_TEST(DirectedPolygonDistanceUnsupportedTypesTest, InvalidTypeTest)
+{
+  using T = TypeParam;
+
+  auto x             = fixed_width_column_wrapper<T>({0, 0});
+  auto y             = fixed_width_column_wrapper<T>({0, 0});
+  auto space_offsets = fixed_width_column_wrapper<size_type>({0});
+
+  EXPECT_THROW(cuspatial::directed_polygon_distance(x, y, space_offsets), cuspatial::logic_error);
+}
```

Comment:
I know this is the perennial discussion, but we should have some tests of non-trivial size (large enough to launch multiple thread blocks).   You could generate a thousand squares on a regular grid (for example) which would make it easy to generate a reference solution analytically.

Or maybe we should look at having gtests run a Python script to generate input and reference data using another library (if that's easier).

Reply:
Opened an issue for good measure. I'll think on how we can approach this.

https://github.com/rapidsai/cuspatial/issues/259